### ISSUE TITLE
Failed to create .htpasswd

### ISSUE THREAD

**Issue by micmicsuarez** [sous-chefs/htpasswd#31]
Hi @guilhem,

I used the example below that you provided in the documentation.
 
```
htpasswd "/etc/nginx/.htpassword" do
  user "admin"
  password "password"
end
```
It seems that it didn't create a .htpassword after the execution of `kitchen verify` command. I used CentOS 7.2.

These are the error logs:
```bash
[2017-06-20T13:19:54+08:00] FATAL: Stacktrace dumped to /tmp/kitchen/cache/chef-stacktrace.out
[2017-06-20T13:19:54+08:00] FATAL: Stacktrace dumped to /tmp/kitchen/cache/chef-stacktrace.out
[2017-06-20T13:19:54+08:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
[2017-06-20T13:19:54+08:00] FATAL: Please provide the contents of the stacktrace.out file if you file a bug report
[2017-06-20T13:19:54+08:00] ERROR: cannot load such file -- htauth
[2017-06-20T13:19:54+08:00] ERROR: cannot load such file -- htauth

```
I used this temporary solution.
```
execute 'htpasswd' do
  command "htpasswd -c -b /etc/nginx/.htpassword admin password"
  not_if { File.exist?('/etc/nginx/.htpassword')}
end
```

Thanks,
Michael

**Comment by guilhem** [comment]
Hi @micmicsuarez 

As indicated, can you provide you `stacktrace.out` for debug?

**Comment by tomwganem** [comment]
Hello,

I am running into the same issue
```
Recipe: <Dynamically Defined Resource>
    * htpasswd[/etc/nginx/slave/elasticsearch-htpasswd] action add

      ================================================================================
      Error executing action `add` on resource 'htpasswd[/etc/nginx/slave/elasticsearch-htpasswd]'
      ================================================================================

      LoadError
      ---------
      cannot load such file -- htauth

      Cookbook Trace:
      ---------------
      /var/chef/cache/cookbooks/htpasswd/providers/default.rb:60:in `initialize'
      /var/chef/cache/cookbooks/aspera-mesos/recipes/_slave_nginx.rb:176:in `block (4 levels) in from_file'
      /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:78:in `run_action'
      /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block (2 levels) in converge'
      /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `each'
      /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block in converge'
      /var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:105:in `converge'

      Compiled Resource:
      ------------------
      # Declared in

      htpasswd("/etc/nginx/slave/elasticsearch-htpasswd") do
        action [:add]
        retries 0
        retry_delay 2
        default_guard_interpreter :default
        user "..."
        password "..."
        not_if { #code block }
      end

      Platform:
      ---------
      x86_64-linux

```
Here's my stacktrace
```
ganem@files-prod-wdc04-agent-80fb94b1-281f-6df0-962f-e590a319db46:~$ sudo cat /var/chef/cache/chef-stacktrace.out
Generated at 2017-06-22 21:40:02 +0000
LoadError: ruby_block[create htpassword elasticsearch-htpasswd] (aspera-mesos::_slave_nginx line 158) had an error: LoadError: htpasswd[/etc/nginx/slave/elasticsearch-htpasswd] (dynamically defined) had an error: LoadError: cannot load such file -- htauth
/opt/chef/embedded/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in `require'
/opt/chef/embedded/lib/ruby/site_ruby/2.1.0/rubygems/core_ext/kernel_require.rb:55:in `require'
/var/chef/cache/cookbooks/htpasswd/providers/default.rb:60:in `initialize'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/resource.rb:1441:in `new'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/resource.rb:1441:in `provider_for_action'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/resource.rb:591:in `run_action'
/var/chef/cache/cookbooks/aspera-mesos/recipes/_slave_nginx.rb:176:in `block (4 levels) in from_file'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/provider/ruby_block.rb:35:in `call'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/provider/ruby_block.rb:35:in `block in action_run'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/mixin/why_run.rb:52:in `add_action'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/provider.rb:176:in `converge_by'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/provider/ruby_block.rb:34:in `action_run'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/provider.rb:145:in `run_action'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/resource.rb:591:in `run_action'
/var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:78:in `run_action'
/var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block (2 levels) in converge'
/var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `each'
/var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:106:in `block in converge'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/resource_collection/resource_list.rb:94:in `block in execute_each_resource'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/resource_collection/stepable_iterator.rb:116:in `call'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/resource_collection/stepable_iterator.rb:116:in `call_iterator_block'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/resource_collection/stepable_iterator.rb:85:in `step'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/resource_collection/stepable_iterator.rb:104:in `iterate'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/resource_collection/stepable_iterator.rb:55:in `each_with_index'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/resource_collection/resource_list.rb:92:in `execute_each_resource'
/var/chef/cache/cookbooks/compat_resource/files/lib/chef_compat/monkeypatches/chef/runner.rb:105:in `converge'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/client.rb:669:in `block in converge'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/client.rb:664:in `catch'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/client.rb:664:in `converge'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/client.rb:703:in `converge_and_save'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/client.rb:283:in `run'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/application.rb:286:in `block in fork_chef_client'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/application.rb:274:in `fork'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/application.rb:274:in `fork_chef_client'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/application.rb:239:in `block in run_chef_client'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/local_mode.rb:44:in `with_server_connectivity'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/application.rb:227:in `run_chef_client'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/application/client.rb:456:in `sleep_then_run_chef_client'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/application/client.rb:443:in `block in interval_run_chef_client'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/application/client.rb:442:in `loop'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/application/client.rb:442:in `interval_run_chef_client'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/application/client.rb:426:in `run_application'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/lib/chef/application.rb:59:in `run'
/opt/chef/embedded/lib/ruby/gems/2.1.0/gems/chef-12.12.15/bin/chef-client:26:in `<top (required)>'
/usr/bin/chef-client:52:in `load'
/usr/bin/chef-client:52:in `<main>'
```

**Comment by guilhem** [comment]
are you using a `chef_version '>= 12.8`' ?

**Comment by gmccue** [comment]
@guilhem I have the same issue on Chef `13.1`. Is the cookbook only compatible with `<= 12.8`?

**Comment by gmccue** [comment]
FYI, I fixed this issue by adding the following to my wrapper recipe...

```
node.override['htpasswd']['install_method'] = 'ruby'

include_recipe 'htpasswd::default'
```

**Comment by tomwganem** [comment]
@guilhem I'm using chef 12.12.15

I've pinned the htpasswd cookbook to 0.2.4 for the time being.

**Comment by predominant** [comment]
@gmccue  Or you can `chef gem 'htauth'` in your wrapper cookbook.

**Comment by ramereth** [comment]
I believe this has been addressed in #41.
